dashboard/app/models/concerns/partial_registration.rb

```diff
@@ -7,62 +7,53 @@
 module PartialRegistration
   extend ActiveSupport::Concern
 
-  USER_ATTRIBUTES_SESSION_KEY = 'devise.user_attributes'
-  OAUTH_PARAMS_TO_STRIP = %w(oauth_token oauth_refresh_token)
+  SESSION_KEY = 'partial_registration'
 
   module ClassMethods
-    def new_from_partial_registration(session)
+    def new_from_partial_registration(session, &block)
       raise 'No partial registration was in progress' unless PartialRegistration.in_progress? session
-      new(session[USER_ATTRIBUTES_SESSION_KEY]) do |user|
-        cache = CDO.shared_cache
-        OAUTH_PARAMS_TO_STRIP.each do |param|
-          next if user.send(param)
-          # Grab the oauth token from memcached if it's there
-          cache_key = PartialRegistration.cache_key(param, user)
-          user.send("#{param}=", cache.read(cache_key)) if cache
-        end
-        yield user if block_given?
-      end
+      cache_key = session[SESSION_KEY]
+      json = CDO.shared_cache.read(cache_key)
+      attributes = JSON.parse(json)
+      new(attributes, &block)
     end
   end
 
   def self.in_progress?(session)
-    !!session[USER_ATTRIBUTES_SESSION_KEY]
+    session[SESSION_KEY] && CDO.shared_cache.exist?(session[SESSION_KEY])
```

Comment:
Will this continue to return true once registration is completed? I see that we delete `session[SESSION_KEY]` if we `cancel`, but I don't see where this would get deleted once registration is complete.

Reply:
Hmm... that's true, I'm not cleaning up a partial registration upon successful registration.  In practice I hadn't thought of a bug that this would immediately cause.  Now that you put it this way, though, it is surprising behavior that `in_progress?` would be true for a registered user, and it's probably worth establishing this cleanup step now.  I'll sort that out and update the PR.

Reply:
Change pushed.  The system now clears any partial registration in progress from both the session and the shared cache upon successful registration.